core/src/main/java/org/apache/carbondata/core/bloom/BloomFilterUtil.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.carbondata.core.bloom;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.BitSet;
+
+import org.apache.carbondata.core.constants.CarbonCommonConstants;
+import org.apache.carbondata.core.constants.CarbonV3DataFormatConstants;
+
+import org.apache.hadoop.util.bloom.BloomFilter;
+import org.roaringbitmap.RoaringBitmap;
+
+public class BloomFilterUtil {
+
+  /**
+   *  Calculate page bloom parameters by item size and fpp
+   *  Formula:
+   *     Number of bits (m) = -n*ln(p) / (ln(2)^2)
+   *     Number of hashes(k) = m/n * ln(2)
+   *
+   * @param n Number of items in the filter
+   * @param p False positive probability
+   *
+   */
+  public static int[] getBloomParameters(int n, double p) {
+    double numOfBit = -n * Math.log(p) / (Math.pow(Math.log(2), 2));
+    double numOfHash = numOfBit / n * Math.log(2);
+    return new int[]{(int) Math.ceil(numOfBit), (int) Math.ceil(numOfHash)};
+  }
+
+  public static int[] getPageBloomParameters() {
+    return getBloomParameters(
+            CarbonV3DataFormatConstants.NUMBER_OF_ROWS_PER_BLOCKLET_COLUMN_PAGE_DEFAULT,
```

Comment:
we have `table_page_size_inmb` table properties, with that column page will not always have 3200 rows, it can have less than that also. 
Number of rows in a page is stored in thrift, `BlockletInfo3.row_count_in_page`. Need to create bloom vector based on this ?

Reply:
You're right. I did try dynamic bloom parameter by dynamic size before. The benefit of setting the maximum pagesize(32000) are:
1. avoid problem of multiple parameters choosing as bloom datamap, only FPP needed to be set;
2. same parameter of bloom has same hash function, so that we only need to hash one time then we can check all pages in blocklet;